indico/cli/i18n.py

```diff
@@ -113,79 +248,261 @@ def _wrapper(**kwargs):
+        module_path = os.path.join(directory, packages[0], 'modules')
+        if not os.path.exists(client_path) and not os.path.exists(module_path):
+            return
+    with _chdir(INDICO_DIR):
+        output = subprocess.check_output(['npx', 'react-jsx-i18n', 'extract', client_path, module_path],
+                                         env=dict(os.environ, FORCE_COLOR='1'))
+    with _chdir(directory):
+        messages_react_pot = _get_react_messages_pot(directory)
+        with open(messages_react_pot, 'wb') as f:
+            f.write(output)
 
-        default = DEFAULT_OPTIONS.get(cmd_name, {}).get(var_name)
-        is_flag = not opt.endswith('=')
-        cmd = click.option(*(opts + [var_name]), is_flag=is_flag, default=default, help=description)(cmd)
-    return cmd
 
+@cli.group('compile', short_help='Catalog compilation command for indico and indico plugins.')
+def compile_catalog():
+    """Catalog compilation command for indico and indico plugins."""
 
-cmd_list = ['init_catalog', 'extract_messages', 'update_catalog', 'compile_catalog',
-            'init_catalog_js', 'extract_messages_js', 'update_catalog_js',
-            'init_catalog_react', 'update_catalog_react']
 
+@cli.group('extract', short_help='Message extraction command for indico and indico plugins.')
+def extract_messages():
+    """Message extraction command for indico and indico plugins."""
 
-for cmd_name in cmd_list:
-    cli.add_command(_make_command(cmd_name))
 
+@cli.group('init', short_help='New catalog initialization command for indico and indico plugins.')
+def init_catalog():
+    """Catalog initialization command for indico and indico plugin.
 
-@cli.command()
-def extract_messages_react():
-    output = subprocess.check_output(['npx', 'react-jsx-i18n', 'extract', 'indico/web/client/', 'indico/modules/'],
-                                     env=dict(os.environ, FORCE_COLOR='1'))
-    with open(MESSAGES_REACT_POT, 'wb') as f:
-        f.write(output)
+    The sub commands take the new locale to be initialized as argument.
+    """
 
 
-@cli.command()
-def compile_catalog_react():
-    for locale in os.listdir(TRANSLATIONS_DIR):
-        po_file = os.path.join(TRANSLATIONS_DIR, locale, 'LC_MESSAGES', 'messages-react.po')
-        json_file = os.path.join(TRANSLATIONS_DIR, locale, 'LC_MESSAGES', 'messages-react.json')
-        if not os.path.exists(po_file):
-            continue
-        output = subprocess.check_output(['npx', 'react-jsx-i18n', 'compile', po_file])
-        json.loads(output)  # just to be sure the JSON is valid
-        with open(json_file, 'wb') as f:
-            f.write(output)
+@cli.group('update', short_help='Catalog merging command for indico and indico plugins.')
+def update_catalog():
+    """Catalog merging command for indico and indico plugins."""
+
+
+def _indico_command(ctx, babel_cmd, python, javascript, react, locale, no_check):  # pylint: disable=too-many-arguments
```

Comment:
We do not use pylint in the core

```suggestion
def _indico_command(ctx, babel_cmd, python, javascript, react, locale, no_check):
```